tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -351,6 +352,20 @@ class AzureEventGridMode(AzureFunctionMode):
                                required=['events'],
                                rinherit=AzureFunctionMode.schema)
 
+    def __init__(self, policy):
+        super(AzureEventGridMode, self).__init__(policy)
+        self.subscribed_events = AzureEvents.get_event_operations(
+            self.policy.data['mode'].get('events'))
+
+    def validate(self):
+        super(AzureEventGridMode, self).validate()
+        resource_type = self.policy.resource_manager.resource_type.resource_type
+        for event in self.subscribed_events:
```

Comment:
Are you going to work on certain child resources next? (Like KeyVault Keys, Storage Account Containers etc).. If not, do you think we should filter them out there?

Reply:
It's still unclear to me - the problem is they can't be the resource. So it's not the fact that it's a child resource, it's just they are non-ARM resources, right? 

And it's not them matching any event, they just can't be set as the `resource` in the policy?

To me these are 2 separate validation bugs.

Reply:
Yeah, this is mostly because they're non-arm which means they might not have any 'type' field at all (not sure if they can still generate IAM events). 

E.g. KeyVault key has no type and its ID is custom

Reply:
oh, non-arm might not support it at all: https://docs.microsoft.com/en-us/azure/event-grid/overview